flexget/components/sites/sites/filelist_api.py

```diff
@@ -0,0 +1,166 @@
+    """
+
+    api_url = 'https://filelist.io/api.php'
+
+    valid_categories = {
+        'Anime': '24',
+        'Audio': '11',
+        'Desene': '15',
+        'Diverse': '18',
+        'Docs': '16',
+        'FLAC': '5',
+        'Filme 3D': '25',
+        'Filme 4K': '6',
+        'Filme 4K Blu-Ray': '26',
+        'Filme Blu-Ray': '20',
+        'Filme DVD': '2',
+        'Filme DVD-RO': '3',
+        'Filme HD': '4',
+        'Filme HD-RO': '19',
+        'Filme SD': '1',
+        'Jocuri Console': '10',
+        'Jocuri PC': '9',
+        'Linux': '17',
+        'Mobile': '22',
+        'Programe': '8',
+        'Seriale 4K': '27',
+        'Seriale HD': '21',
+        'Seriale SD': '23',
+        'Sport': '13',
+        'TV': '14',
+        'Videoclip': '12',
+        'XXX': '7',
+    }
+
+    valid_extras = {'internal': 1, 'moderated': 1, 'freeleech': 1, 'doubleup': 1}
+    schema = {
+        'type': 'object',
+        'properties': {
+            'username': {'type': 'string'},
+            'passkey': {'type': 'string'},
+            'category': one_or_more({'type': 'string', 'enum': list(valid_categories.keys())}),
+            'internal': {'type': 'boolean', 'default': False},
+            'moderated': {'type': 'boolean', 'default': False},
+            'freeleech': {'type': 'boolean', 'default': False},
+            'doubleup': {'type': 'boolean', 'default': False},
+        },
+        'required': ['username', 'passkey'],
+        'additionalProperties': False,
+    }
+
+    def get(self, url, params):
+
+        try:
+            response = requests.get(url, params=params, raise_status=False)
+        except RequestException as e:
+            raise plugin.PluginError(f'FileList request failed badly! {e}')
+
+        if not response.ok:
+            http_status = HTTPStatus(response.status_code)
+            error_message = response.json().get('error', http_status.description)
+
+            raise plugin.PluginError(f'FileList request failed; err code: {http_status}; err msg: `{error_message}`')
+
+        return response
+
+    @plugin.internet(logger)
+    def search(self, task, entry, config):
+        """
+            Search for entries on FileList
+        """
+        entries = []
+
+        # TODO remove later
+        logger.debug(entry)
+
+        # mandatory params
+        params = {'username': config['username'], 'passkey': config['passkey'], 'action': 'search-torrents'}
+
+        # category
+        if config.get('category'):
+            params['category'] = (
+                ','.join(self.valid_categories[cat] for cat in config['category'])
+                if isinstance(config.get('category'), list)
+                else self.valid_categories[config.get('category')]
+            )
+
+        # extras: internal release, moderated torrent, freeleech
+        for extra_name, extra_value in self.valid_extras.items():
+            if config.get(extra_name):
+                params[extra_name] = extra_value
```

Comment:
seems a bit redundant to hold these "extra" in an additional dict, espicially since their values is always one. consider this:
```suggestion
        params.update({extra: 1 for extra in ('internal', 'moderated', 'freeleech', 'doubleup') if config.get(extra)})
```

Reply:
while true the value of the extras are always one, i will still like to keep it this ways since if they introduce more extras with different values i will just need to update the dict.

Reply:
the `1` is just simple way to pass `True` via params, so this value will probably not change. How about this for a compromise, set the class attribute to hold a tuple of the names and iterate over those

Reply:
willing to take a list over a tuple ? 😄 if yes, all required changes are committed